pkg/root/trusted_root.go

```diff
@@ -145,32 +153,54 @@ func ParseTransparencyLogs(tlogs []*prototrustroot.TransparencyLogInstance) (tra
 			var ecKey *ecdsa.PublicKey
 			var ok bool
 			if ecKey, ok = key.(*ecdsa.PublicKey); !ok {
-				return nil, fmt.Errorf("tlog public key is not ECDSA P256")
+				return nil, fmt.Errorf("tlog public key is not ECDSA: %s", tlog.GetPublicKey().GetKeyDetails())
+			}
+			tlogEntry.PublicKey = ecKey
+			tlogEntry.SignatureHashFunc = crypto.SHA256
+		// This key format has public key in PKIX RSA format and PKCS1#1v1.5 or RSASSA-PSS signature
+		case protocommon.PublicKeyDetails_PKIX_RSA_PKCS1V15_2048_SHA256,
+			protocommon.PublicKeyDetails_PKIX_RSA_PKCS1V15_3072_SHA256,
+			protocommon.PublicKeyDetails_PKIX_RSA_PKCS1V15_4096_SHA256,
+			protocommon.PublicKeyDetails_PKIX_RSA_PSS_2048_SHA256,
+			protocommon.PublicKeyDetails_PKIX_RSA_PSS_3072_SHA256,
+			protocommon.PublicKeyDetails_PKIX_RSA_PSS_4096_SHA256:
+			key, err := x509.ParsePKIXPublicKey(tlog.GetPublicKey().GetRawBytes())
+			if err != nil {
+				return nil, err
 			}
-			transparencyLogs[encodedKeyID] = &TransparencyLog{
-				BaseURL:           tlog.GetBaseUrl(),
-				ID:                tlog.GetLogId().GetKeyId(),
-				HashFunc:          hashFunc,
-				PublicKey:         ecKey,
-				SignatureHashFunc: crypto.SHA256,
+			var rsaKey *rsa.PublicKey
+			var ok bool
+			if rsaKey, ok = key.(*rsa.PublicKey); !ok {
+				return nil, fmt.Errorf("tlog public key is not RSA: %s", tlog.GetPublicKey().GetKeyDetails())
 			}
+			tlogEntry.PublicKey = rsaKey
+			tlogEntry.SignatureHashFunc = crypto.SHA256
+		case protocommon.PublicKeyDetails_PKIX_ED25519, protocommon.PublicKeyDetails_PKIX_ED25519_PH:
+			key, err := x509.ParsePKIXPublicKey(tlog.GetPublicKey().GetRawBytes())
+			if err != nil {
+				return nil, err
+			}
+			var edKey *ed25519.PublicKey
+			var ok bool
+			if edKey, ok = key.(*ed25519.PublicKey); !ok {
+				return nil, fmt.Errorf("tlog public key is not RSA: %s", tlog.GetPublicKey().GetKeyDetails())
+			}
+			tlogEntry.PublicKey = edKey
+			tlogEntry.SignatureHashFunc = crypto.SHA512 // see RFC8032
```

Comment:
Is `hashFunc` related to `SignatureHashFunc`? ED25519 uses SHA512 signature scheme and only SHA256 is supported as a hash function: https://github.com/sigstore/sigstore-go/blob/3a6a109900dfe00c55d51d18556c37859fbae3b8/pkg/root/trusted_root.go#L132-L137